### PR TITLE
fix: various custom operator conformance fixes

### LINKED ISSUE #634: Update flagd-testbed to v3.6.2 and fix in-process evaluation edge cases
## Summary

Update the flagd-testbed dependency to [v3.6.2](https://github.com/open-feature/flagd-testbed/releases/tag/v3.6.2) and fix the in-process evaluation to pass all new edge-case tests.

Testbed v3.6.2 adds coverage for the edge cases identified in:
- open-feature/flagd#1873 (sem_ver parsing inconsistencies)
- open-feature/flagd#1874 (custom operator error return values)
- open-feature/flagd#1875 ($evaluators/$ref resolution gaps)

### sem_ver (open-feature/flagd#1873)

1. **sem_ver rejects `v`-prefixed versions** (e.g. `"v1.2.3"`); `SemVersionStyles.Strict` does not allow `v` prefix; should strip `v`/`V` prefix before parsing
2. **sem_ver uses `CompareSortOrderTo()` instead of `ComparePrecedenceTo()`**; build metadata is included in comparisons, violating SemVer 2.0.0 Rule 10 (e.g. `"1.0.0+build1" == "1.0.0"` returns `false` instead of `true`)
3. **sem_ver returns `false` on parse failure instead of `null`**; should return `null` to trigger default variant fallback
4. **sem_ver returns `false` on unknown operator instead of `null`**; same issue as above

### Custom operator error returns (open-feature/flagd#1874)

5. **`starts_with` returns `false` on error instead of `null`**; should return `null`
6. **`ends_with` returns `false` on error instead of `null`**; should return `null`
7. **`fractional` returns `""` (empty string) on no-bucket-match**; should return `null` to trigger default variant fallback

### $evaluators/$ref resolution (open-feature/flagd#1875)

8. **Non-deterministic `$evaluators` replacement ordering** due to `Dictionary` having no guaranteed iteration order; should use deterministic iteration
9. **No diagnostic feedback on unresolved `$ref`s at parse time**; failure only surfaces at evaluation time
10. **$ref regex relies on `JsonSerializer.Serialize()` output format**; no tolerance for whitespace variations in input

Parent: open-feature/flagd#1904

### PR DESCRIPTION
**Pull request by toddbaert**
fix: various custom operator conformance fixes

* support v/V prefix in semver
* support partial versions in semver
* ignore build metadata in semver comparison
* return null on errors
* handle unresolved $ref as evaluation error
* fix fractional with missing targetingKey

:warning: wait until flagd/RPC is updated to merge.

Fixes: https://github.com/open-feature/dotnet-sdk-contrib/issues/634

---

Related:
* https://github.com/open-feature/flagd/pull/1950
* https://github.com/open-feature/java-sdk-contrib/pull/1778
* https://github.com/open-feature/js-sdk-contrib/pull/1519
* https://github.com/open-feature/python-sdk-contrib/pull/386
* https://github.com/open-feature/go-sdk-contrib/pull/874

### CODE REVIEW

File: src/OpenFeature.Providers.Flagd/Resolver/InProcess/JsonEvaluator.cs [hunk 1]
```diff
@@ -81,7 +82,7 @@ internal FlagSyncData Parse(string flagConfigurations)
             parsed.Evaluators.Keys.ToList().ForEach(key =>
             {
                 var val = parsed.Evaluators[key];
-                var evaluatorRegex = new Regex("{\"\\$ref\":\"" + key + "\"}");
+                var evaluatorRegex = new Regex("{\"\\$ref\"\\s*:\\s*\"" + key + "\"}");
                 transformed = evaluatorRegex.Replace(transformed, Convert.ToString(val));
             });
```
[the remark this answers is not in the document]
Comment (toddbaert) [reply src/OpenFeature.Providers.Flagd/Resolver/InProcess/JsonEvaluator.cs]: All valid. Fixed. Unit test added.

**Review by kylejuliandev — APPROVED** [review]

File: src/OpenFeature.Providers.Flagd/Resolver/InProcess/CustomEvaluators/FractionalRule.cs [hunk 2]
```diff
@@ -42,6 +42,10 @@ public JsonNode Apply(JsonNode args, EvaluationContext context)
         }
         else
         {
+            if (string.IsNullOrEmpty(flagdProperties.TargetingKey))
```
Comment (askpt) [inline src/OpenFeature.Providers.Flagd/Resolver/InProcess/CustomEvaluators/FractionalRule.cs]: Are whitespaces allowed? This only validates `null` and `""`. Should we include `"  "`

File: src/OpenFeature.Providers.Flagd/Resolver/InProcess/CustomEvaluators/UnresolvedRefRule.cs [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+/// <summary>
+/// Catch-all rule for unresolved $ref references.
+/// If a $ref was not replaced during the evaluator transform step, this rule throws so the caller gets a PARSE_ERROR.
+/// This behavior is consistent with other implementations (though the mechanism differs).
+/// </summary>
+internal sealed class UnresolvedRefRule : IRule
+{
+    public JsonNode Apply(JsonNode args, EvaluationContext context)
+    {
+        var evaluatorName = args?.ToString() ?? "unknown";
```
Comment (askpt) [inline src/OpenFeature.Providers.Flagd/Resolver/InProcess/CustomEvaluators/UnresolvedRefRule.cs]: Should "unknown" be a `const`?

File: src/OpenFeature.Providers.Flagd/Resolver/InProcess/JsonEvaluator.cs [hunk 4]
```diff
@@ -78,12 +79,14 @@ internal FlagSyncData Parse(string flagConfigurations)
...
         {
-            parsed.Evaluators.Keys.ToList().ForEach(key =>
+            foreach (var key in parsed.Evaluators.Keys)
             {
                 var val = parsed.Evaluators[key];
-                var evaluatorRegex = new Regex("{\"\\$ref\":\"" + key + "\"}");
-                transformed = evaluatorRegex.Replace(transformed, Convert.ToString(val));
-            });
+                var escapedKey = Regex.Escape(key);
+                var evaluatorRegex = new Regex("{\"\\$ref\"\\s*:\\s*\"" + escapedKey + "\"}");
```
Comment (askpt) [inline src/OpenFeature.Providers.Flagd/Resolver/InProcess/JsonEvaluator.cs]: Should we consider caching the compiled regexes? The tradeoff is the initial Regex will take time, but the subsequent will be faster.

```suggestion
                var evaluatorRegex = new Regex("{\"\\$ref\"\\s*:\\s*\"" + escapedKey + "\"}", RegexOptions.Compiled);
```

**Review by askpt — COMMENTED** [review]
Added a couple of comments. Other than that LGTM.

File: src/OpenFeature.Providers.Flagd/Resolver/InProcess/CustomEvaluators/FractionalRule.cs (hunk 2 above)
Comment (toddbaert) [reply src/OpenFeature.Providers.Flagd/Resolver/InProcess/CustomEvaluators/FractionalRule.cs]: Good call.

File: src/OpenFeature.Providers.Flagd/Resolver/InProcess/JsonEvaluator.cs (hunk 4 above)
Comment (toddbaert) [reply src/OpenFeature.Providers.Flagd/Resolver/InProcess/JsonEvaluator.cs]: I guess it will also take a bit of memory, but I think it's definitely better. Good idea. TBH I had no idea about this option.

**Comment by toddbaert** [comment]
@askpt Thanks. I made your recommended changes. Would you mind looking at the equivalent in [flagd](https://github.com/open-feature/flagd/pull/1950)? We need these changes there as well, so remote and in-process evaluations behave identically.

**Review by askpt — APPROVED** [review]
